uwblueprint/humane-society
# Private Routes With Roles

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)



## Implementation description
* Updated private route functionality to support roles.



## Steps to test
1. Give yourself a role in your local DB (You can check the roles table for the different roles)
2. Delete existing cached user information (You can do this in dev tools. Click the 'Application' tab and go to 'Local Storage')
3. Refresh page and log back in.
4. Try to access different pages and see if you are able to access with the allowed role(s), and if you are restricted if you don't have the right role(s). (Check App.tsx) for allowed roles per page)



## What should reviewers focus on?
* Currently the roles allowed per page are just dummy data, this will need to be updated once we solidify the role restrictions per page


## Checklist
